src/ethernet_interface.cpp

```diff
@@ -319,6 +319,20 @@ ObjectPath EthernetInterface::ip(IP::Protocol protType, std::string ipaddress,
 
     std::string objectPath =
         generateObjectPath(protType, ipaddress, prefixLength, gateway);
+
+    auto it = this->addrs.find(ipaddress);
+    if (it != this->addrs.end())
+    {
+        if (it->second->getObjPath() == objectPath)
+        {
+            return it->second->getObjPath();
+        }
+        else
+        {
+            this->addrs.erase(it);
```

Comment:
Is this else code block ever gets hit? What is erased here?

Reply:
> Is this else code block ever gets hit? What is erased here?

Yes when we try to add an existing static IP but with the modified Gateway and subnet.
Example: initial config:  
{"Address": "9.3.29.25", "SubnetMask": "255.255.255.0","Gateway":"9.3.29.1"}

New Request:
{"Address": "9.3.29.25", "SubnetMask": "255.255.0.0","Gateway":"9.3.29.1"}